co-simulation/fed/mosaic-carla/src/main/java/org/eclipse/mosaic/fed/carla/carlaconnect/CarlaXmlRpcClient.java

```
@@ -0,0 +1,112 @@
+    private XmlRpcClientConfigImpl config;
+    private final Logger log = LoggerFactory.getLogger(this.getClass());
+
+    public CarlaXmlRpcClient() {
+        
+    }
+
+    /**
+     * This method is used to send a stop message to the python server side to shut down server from there
+     */
+    public void closeConnection()
+    {
+        log.info("carla connection server closing");
+        isConnected = false;
+        try {
+            Object[] params = new Object[]{"Stop Server"};
+            String result =  (String)client.execute(registeredFunction, params);
+            log.info(result);
+        } 
+        catch (XmlRpcException e) 
+        {
+            log.error("Errors occurred with {0}", e.getMessage());
+        }
+    }
+
+    public void initialize()
+    {
+        try{           
+            config = new XmlRpcClientConfigImpl();
+            config.setServerURL(new URL("http://127.0.0.1:8090/RPC2"));
+            client = new XmlRpcClient();
+            client.setConfig(config);
+            Object[] connectionTest = new Object[]{"Connection test"};
+            Object result = (String)client.execute(registeredFunction, connectionTest);
+            isConnected = true;
+
+            if(!result.equals("Connected!"))
+            {
+               isConnected = false;
+               log.error("Server is not connected!");
+            }
+        }
+        catch (MalformedURLException m) 
+        {
+            log.error("Errors occurred with {0}", m.getMessage());
+            isConnected = false;
+        }
+        catch (XmlRpcException x) 
+        {
+            log.error("Errors occurred with xmlrpc connection {0}", x.getMessage());
+            isConnected = false;
+        } 
+
+    }
+
+
+    /**
+    * This method uses xmlrpc to connect with CARLA CDASim adapter service
+    */
+    public void requestCarlaList()
+    {
+        if(!isConnected)
+        {
+            log.error("Server is not connected!");
```

throw an exception instead of logging error and returning